### PR TITLE
SDK: integrate AggregateBalanceObserver

### PR DESCRIPTION
**Pull request by gasiortomasz**

### CODE REVIEW

**Comment by marekkirejczyk** [comment]
Need rebase @gasiortomasz

File: universal-login-sdk/lib/api/sdk.ts [hunk 1]
```diff
@@ -105,16 +110,17 @@ class UniversalLoginSDK {
...
+    if (this.balanceObserver && this.aggregateBalanceObserver) {
+      return;
     }
     const walletContractAddress = await this.getWalletContractAddress(ensName);
     ensureNotNull(walletContractAddress, InvalidContract);
     ensureNotNull(this.relayerConfig, MissingConfiguration);
 
     const tokenDetails = await this.getTokensDetails();
     this.balanceObserver = new BalanceObserver(this.balanceChecker, walletContractAddress, tokenDetails);
+    this.aggregateBalanceObserver = new AggregateBalanceObserver(this.balanceObserver!, this.priceOracle);
```
Comment (JustynaBroniszewska) [inline universal-login-sdk/lib/api/sdk.ts]: There is a possibility that we only use balanceObserver without aggregateBalanceObserver, for example: [subscribeToBalances](https://github.com/UniversalLogin/UniversalLoginSDK/pull/752/files#diff-ddd14f831e8efeac800fe4a489d7fbe0L212)

File: universal-login-sdk/test/e2e/AggregateBalanceObserver.ts [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+import {Wallet, utils} from 'ethers';
+import {createFixtureLoader} from 'ethereum-waffle';
+import {waitUntil} from '@universal-login/commons';
+import {RelayerUnderTest} from '@universal-login/relayer';
+import UniversalLoginSDK from '../../lib/api/sdk';
+import basicSDK from '../fixtures/basicSDK';
+
+const loadFixture = createFixtureLoader();
+
+describe('E2E: AggregateBalanceObserver', () => {
```
Comment (marekkirejczyk) [inline universal-login-sdk/test/e2e/AggregateBalanceObserver.ts]: Feels like unnecessary tests, we already have units.

File: universal-login-sdk/test/e2e/BalanceObserver.ts [hunk 3]
```diff
@@ -34,7 +33,6 @@ describe('E2E: BalanceObserver', () => {
   it('1 subscription - balance changed', async () => {
     const callback = sinon.spy();
 
-    await sdk.fetchBalanceObserver(ensName);
     const unsubscribe = await sdk.subscribeToBalances(ensName, callback);
```
Comment (marekkirejczyk) [inline universal-login-sdk/test/e2e/BalanceObserver.ts]: In general, I don't feel we need e2e tests for Domain services. 
I would propose unit tests for them + integration tests for the integration layer and e2e for the sdk.

**Pull request merged**